src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -182,6 +183,13 @@ applyBlocks blocks cp0 =
     applyBlock' (txs, cp) b =
         let (txs', cp') = applyBlock b cp in (txs <> txs', cp')
 
+newPending
+    :: Tx
+    -> Wallet s
+    -> Wallet s
+newPending !tx (Wallet !utxo !pending !history !tip !s) =
```

Comment:
maybe dumb question: is there a reason strict annotation is really necessary here? (I don't usually use this until I am really sure this will create lots of unnecessary thunks - but I haven't looked at the code to figure out is this so)

Reply:
On `newPending` it's more debatable. It's definitely necessary on the `applyBlock` to avoid constructing too many chunks (as we generally apply a whole epoch before evaluating the state again). On new pending, we evaluate it right away by storing it to disk.